Sources/Public/CalendarView.swift

```diff
@@ -617,43 +613,43 @@ public final class CalendarView: UIView {
   }
 
   private func updateVisibleViews(
-    withVisibleItems visibleItems: Set<VisibleCalendarItem>,
-    previouslyVisibleItems: Set<VisibleCalendarItem>)
+    withVisibleItems visibleItems: Set<VisibleItem>,
+    previouslyVisibleItems: Set<VisibleItem>)
   {
-    var visibleItemsWithViewsToRemove = previouslyVisibleItems
-    let previousVisibleViewsForVisibleItems = visibleViewsForVisibleItems
-
+    var viewsToHideForVisibleItems = visibleViewsForVisibleItems
     visibleViewsForVisibleItems.removeAll(keepingCapacity: true)
 
     reuseManager.viewsForVisibleItems(
       visibleItems,
       viewHandler: { view, visibleItem, previousBackingVisibleItem, isReusedViewSameAsPreviousView in
         UIView.conditionallyPerformWithoutAnimation(when: !isReusedViewSameAsPreviousView) {
           if view.superview == nil {
-            scrollView.addSubview(view)
+            subviewsManager.insertSubview(view, correspondingItemType: visibleItem.itemType)
           }
 
+          view.isHidden = false
+
           configureView(view, with: visibleItem)
         }
 
         visibleViewsForVisibleItems[visibleItem] = view
 
         if let previousBackingVisibleItem = previousBackingVisibleItem {
-          visibleItemsWithViewsToRemove.remove(previousBackingVisibleItem)
+          // Don't hide views that were reused
+          viewsToHideForVisibleItems.removeValue(forKey: previousBackingVisibleItem)
         }
       })
 
-    // Remove any old views that weren't reused
-    for visibleItemWithViewToRemove in visibleItemsWithViewsToRemove {
-      previousVisibleViewsForVisibleItems[visibleItemWithViewToRemove]?.removeFromSuperview()
+    // Hide any old views that weren't reused. This is faster than adding / removing subviews.
+    for (_, viewToHide) in viewsToHideForVisibleItems {
+      viewToHide.isHidden = true
     }
```

Comment:
new perf optimization to avoid adding / removing subviews more than needed (we can just hide / unhide)